### PR TITLE
Chore: renamed ring.IngesterDesc to ring.InstanceDesc

### PR DESCRIPTION
**Pull request by pracucci**
**What this PR does**:
Following up the PR https://github.com/cortexproject/cortex/pull/3760, in this PR I've renamed `ring.IngesterDesc` to `ring.InstanceDesc`. No other change has been done.

This should be a safe change, because doesn't affect protobuf serialization and json tags of protobuf generated `struct` hasn't changed.

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by jtlisi — APPROVED** [review]
LGTM, nothing in the diff besides renaming

**Review by khaines — APPROVED** [review]

**Comment by pstibrany** [comment]
Should we tackle `IngesterState` in this PR too?

File: pkg/ring/ring.proto [hunk 1]
```diff
@@ -8,11 +8,11 @@ option (gogoproto.marshaler_all) = true;
 option (gogoproto.unmarshaler_all) = true;
 
 message Desc {
-	map<string,IngesterDesc> ingesters = 1 [(gogoproto.nullable) = false];
+	map<string,InstanceDesc> ingesters = 1 [(gogoproto.nullable) = false];
```
Comment (pstibrany) [inline pkg/ring/ring.proto:11]:
```suggestion
	map<string,InstanceDesc> instances = 1 [(gogoproto.nullable) = false];
```

File: pkg/ring/model.go [hunk 2]
```diff
@@ -32,23 +32,23 @@ func GetCodec() codec.Codec {
...
 	return &Desc{
-		Ingesters: map[string]IngesterDesc{},
+		Ingesters: map[string]InstanceDesc{},
 	}
 }
 
 // AddIngester adds the given ingester to the ring. Ingester will only use supplied tokens,
 // any other tokens are removed.
-func (d *Desc) AddIngester(id, addr, zone string, tokens []uint32, state IngesterState, registeredAt time.Time) IngesterDesc {
+func (d *Desc) AddIngester(id, addr, zone string, tokens []uint32, state IngesterState, registeredAt time.Time) InstanceDesc {
```
Comment (pstibrany) [inline pkg/ring/model.go:41]:
```suggestion
func (d *Desc) AddInstance(id, addr, zone string, tokens []uint32, state IngesterState, registeredAt time.Time) InstanceDesc {
```

**Review by pstibrany — APPROVED** [review]
LGTM. There are still some "ingester" left in ring.proto, and entire ring package, but we can fix it gradually in subsequent PRs.

File: pkg/ring/ring.proto (hunk 1 above)
Comment (pracucci) [reply pkg/ring/ring.proto:11]: Will do in separate PRs.

File: pkg/ring/model.go (hunk 2 above)
Comment (pracucci) [reply pkg/ring/model.go:41]: Planned to do in separate PRs.

**Comment by pracucci** [comment]
> There are still some "ingester" left in ring.proto, and entire ring package, but we can fix it gradually in subsequent PRs.

Will do other renamings in separate PRs in other to keep the diff smaller and easier to review.

**Pull request merged**